utils/text_formatting.jsx

```diff
@@ -417,7 +417,7 @@ export function parseSearchTerms(searchTerm) {
         }
 
         // check for a search flag (and don't add it to terms)
-        captured = (/^(?:in|from|channel): ?\S+/).exec(termString);
+        captured = (/^(?:in|-in|from|-from|channel|-channel): ?\S+/).exec(termString);
```

Comment:
Not sure if this is correct, but something like this?
```suggestion
        captured = (/^-?(?:in|from|channel): ?\S+/).exec(termString);
```